packages/core/src/rum/instrumentation/resourceTracking/implementation/URLHostParser.ts

```diff
@@ -0,0 +1,30 @@
+/*
+ * Unless explicitly stated otherwise all files in this repository are licensed under the Apache License Version 2.0.
+ * This product includes software developed at Datadog (https://www.datadoghq.com/).
+ * Copyright 2016-Present Datadog, Inc.
+ */
+
+import { InternalLog } from '../../../../InternalLog';
+import { SdkVerbosity } from '../../../../SdkVerbosity';
+import type { Hostname } from '../domain/firstPartyHosts';
+
+// matches what is between the first "://" and the next "/", ":" or whitespace
+const hostRegex = '^.+://([^:/\\s]+)';
+
+/**
+ * Returns the host from an URL.
+ * @returns the host (without the port) or null if could not cast the URL as host
+ */
+export const URLHostParser = (url: string): Hostname | null => {
```

Comment:
So this is what is not quite clear to me after reading the RFC: what is the difference between `domain` and `implementation` folders? Do you mean that `domain` should contain Datadog-specific business logic and `implementation` everything else what is not Datadog-related?

Reply:
I think this one should be in the domain, but I am not sure.
Basically I want to separate is:
- the code that is dependent on React, React Native (it might not work after a RN upgrade) or the native SDK 
- the code that is independent from all that, that can run the same way on RN, a browser or jest 

I guess the names are not appropriate here, I choose them because I have in mind the onion architecture (or hexagonal architecture).
We have one application layer which is the point of entry (here `DdRumResourceTracking.tsx`) which depends on RN, then some logic that does not depend on RN (that I refer to as domain layer, but there are some business rules as well), and finally the infrastructure layer which makes calls to RN, DdRUM etc. and that has a higher probability of being modified.

Reply:
Are you ok with leaving it as it is right now, and to come back to it after 1-2 more features?

Reply:
Ok, separation between target-specific (React, React native, etc.) code vs generic code makes sense.

Imo `infrastructure` name is quite confusing here, because I would get this definition as something low level, not high level as you described.

But yes, we can keep it like that for now and come back later, it is a private API anyway.